src/compiler/utilitiesPublic.ts

```diff
@@ -1657,13 +1657,13 @@ export function isFunctionLike(node: Node | undefined): node is SignatureDeclara
 }
 
 /** @internal */
-export function isFunctionLikeOrClassStaticBlockDeclaration(node: Node | undefined): node is SignatureDeclaration | ClassStaticBlockDeclaration {
-    return !!node && (isFunctionLikeKind(node.kind) || isClassStaticBlockDeclaration(node));
+export function isFunctionLikeOrClassStaticBlockDeclaration(node: Node | undefined): node is FunctionLikeDeclaration | ClassStaticBlockDeclaration {

```

Comment:
`isFunctionLikeDeclaration` handles value-space declaration whereas `isFunctionLikeKind` handles value and type space declarations. `isFunctionLikeOrClassStaticBlockDeclaration` always works within the value space so I think it's more appropriate to use `isFunctionLikeDeclaration` here